### PR TITLE
Animate media progress bar

### PR DESCRIPTION
**Pull request by fstanis**
#### WHAT
Makes `MediaControlButtons` and others have animated progress.

#### WHY
Generally provides better UX, especially for shorter media and when moving from one media item to another.

#### HOW
By using `animateFloatAsState` for setting progress.

#### Checklist :clipboard:
- [X] Add explicit visibility modifier and explicit return types for public declarations
- [X] Run spotless check
- [X] Run tests
- [X] Update metalava's signature text files

### CODE REVIEW

File: media-ui/src/main/java/com/google/android/horologist/media/ui/screens/player/PlayerScreen.kt [hunk 1]
```diff
@@ -134,7 +134,8 @@ public fun DefaultPlayerScreenControlButtons(
         onSeekToNextButtonClick = { playerController.skipToNextMedia() },
         seekToNextButtonEnabled = playerUiState.seekToNextEnabled,
         showProgress = playerUiState.trackPosition?.showProgress ?: true,
-        percent = playerUiState.trackPosition?.percent ?: 0f
+        percent = playerUiState.trackPosition?.percent ?: 0f,
+        animateProgress = true
```
Comment (yschimke) [inline media-ui/src/main/java/com/google/android/horologist/media/ui/screens/player/PlayerScreen.kt:138]: We should probably discuss the difference between AnimatedXButtons and XButtons.

Do we want some 

a) simple buttons with no animation
b) buttons with full animation including lottie
c) simple buttons with some (non-lottie) animation

I've been aiming for a+b.

File: media-ui/src/main/java/com/google/android/horologist/media/ui/components/PlayPauseButton.kt [hunk 2]
```diff
@@ -108,10 +113,14 @@ public fun PlayPauseProgressButton(
                 .clip(CircleShape)
                 .background(backgroundColor)
         ) {
+            val progress by animateFloatAsState(
+                targetValue = percent.ifNan(0f),
+                animationSpec = if (animateProgress) ProgressIndicatorDefaults.ProgressAnimationSpec else snap()
```
Comment (yschimke) [inline media-ui/src/main/java/com/google/android/horologist/media/ui/components/PlayPauseButton.kt:118]: Does this default to ?

```
const val DefaultDurationMillis: Int = 300
```

File: media-ui/src/main/java/com/google/android/horologist/media/ui/screens/player/PlayerScreen.kt (hunk 1 above)
Comment (yschimke) [reply media-ui/src/main/java/com/google/android/horologist/media/ui/screens/player/PlayerScreen.kt:138]: Should we remove the animateProgress flag and just go for b+c? Turn this on by default?

Comment (fstanis) [reply media-ui/src/main/java/com/google/android/horologist/media/ui/screens/player/PlayerScreen.kt]: Agreed, removed flag.

**Review by yschimke — APPROVED** [review]

**Comment by fstanis** [comment]
After some discussion, this will probably be superseded completely by #806, so treat it as a stopgap solution towards smoother progress.

**Pull request merged**